plugins/modules/ndo_route_map_policy_route_control_context.py

```diff
@@ -0,0 +1,441 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_route_map_policy_route_control_context
+short_description: Manage Route Map Policy for Route Control Context on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Route Map Policy for Route Control Context on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v4.1 and later.
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a tenant template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the tenant template.
+    - This parameter or O(template) is required.
+    type: str
+  route_map_policy:
+    description:
+    - The name of the Route Map Policy for Route Control.
+    - This parameter or O(route_map_policy_uuid) is required.
+    type: str
+  route_map_policy_uuid:
+    description:
+    - The UUID of the Route Map Policy for Route Control.
+    - This parameter or O(route_map_policy) is required.
+    type: str
+  name:
+    description:
+    - The name of the Route Control context.
+    type: str
+  description:
+    description:
+    - The description of the Route Control context.
+    - Providing an empty string O(description="") will remove description from the Route Control context.
+    type: str
+  order:
+    description:
+    - The order of the Route Control context.
+    - The value must be a number between 0 and 9.
+    - Defaults to C(0) when unset during creation.
+    type: int
+  action:
+    description:
+    - The action of the Route Control context.
+    - Defaults to C(permit) when unset during creation.
+    type: str
+    choices: [ permit, deny ]
+  set_rule_uuid:
+    description:
+    - The UUID of the Set Rule Policy.
+    - Providing an empty string O(set_rule_uuid="") will remove Set Rule Policy from the Route Control context.
+    - This parameter can be used instead of O(set_rule).
+    type: str
+  set_rule:
+    description:
+    - The reference of the Set Rule Policy.
+    - Providing an empty dictionary O(set_rule={}) will remove Set Rule Policy from the Route Control context.
+    - This parameter can be used instead of O(set_rule_uuid).
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the Set Rule Policy.
+        type: str
+      template:
+        description:
+        - The template associated with the Set Rule Policy.
+        - This parameter or O(set_rule.template_id) is required.
+        type: str
+      template_id:
+        description:
+        - The template ID associated with the Set Rule Policy.
+        - This parameter or O(set_rule.template) is required.
+        type: str
```

Comment:
why is this not a uuid and reference as in match_rules? i thought that was the current approach of uuid and reference passing?

Reply:
Yes, changed!